packages/mimir/src/rules/no-duplicate-spread-property.ts

```diff
@@ -81,16 +107,50 @@ export class Rule extends TypedRule {
                     names,
                     known: true,
                     assignedNames: names.length !== 1 ? [] : names, // if the computed name is a union, it's not sure which will be assigned
+                    contextFiltered: false,
                 };
             }
         }
     }
 
     private getPropertyInfoFromSpread(node: ts.Expression): PropertyInfo {
-        return getPropertyInfoFromType(this.checker.getTypeAtLocation(node)!);
+        const spreadProperties = getPropertyInfoFromType(this.checker.getTypeAtLocation(node)!);
+        if (!spreadProperties.known)
+            return spreadProperties;
+        const contextualType = this.checker.getContextualType(node);
```

Comment:
this should be done once for the object literal and not once per SpreadAssignment